[STRATO-2587] Add userAddress-x509CertContract key-value storage to Redis

luketollefson opened the pull request:
@trdancer I changed the `RegisterCertificate` contract for testing purposes. Go ahead and delete my additions/change things around. I just added the lines for testing purposes.

I would recommend merging in https://github.com/blockapps/strato-platform/pull/1523 first.

Add a userAddress-x509CertContract mapping to Redis for X509. The implementation follows the implementation of the `MemberAdded` and `MemberRemoved` events very closely.

Done

luketollefson committed:
Add (k:userAddress,v:x509CertContract) to Redis

Add a userAddress-x509CertContract mapping to Redis for X509.

luketollefson committed:
Merge branch 'Fix-CertificateRegistry-Dapp' into STRATO-2587-X509-RedisDB

trdancer reviewed (commented):
As for tests, it doesn't seem like there are any existing tests for the indexer... Which might be an intential thing, or something that needs to be fixed! But there would probably a bit of setup involved to make the test suite... But if there were tests to be had, I would say some simple ones would be like:

- Inserts the data when a new cert registered event is emitted

luketollefson committed:
Properly index CertificateRegistered events

CertificateRegistered events were only available on private
chains by accident. But now they are properly only available
on the main chain.

luketollefson committed:
Add unit tests for indexEventToTxrResults

luketollefson committed:
Merge branch 'develop' into STRATO-2587-X509-RedisDB

luketollefson committed:
Merge branch 'develop' into STRATO-2587-X509-RedisDB
